guake/tests/test_notebook.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+import mock
+from mock import Mock, MagicMock
+
+import guake.notebook
+
+
+guake.notebook.TerminalBox = Mock()
```

Comment:
This is not how to mock. You need to use mocker.patch or mock.patch to ensure your unit test restore the initial state at the end of the execution.
I would do something like:
@pytest.fixture(scope=module)
def fake_notebook(mocker):
   mocker.patch(....)

def test_something(fake_notebook):
   ...

Reply:
OK, I definitely need to first take a look about pytest documentation 🤔

Reply:
There some good article:
- https://codefellows.github.io/sea-python-401d7/lectures/mock.html
- https://flowfx.de/blog/using-the-pytest-mock-plugin/

You might also need to understand how the [conftest.py trick works](https://docs.pytest.org/en/2.7.3/plugins.html?highlight=re) to refactorize fixture.

Thanks a lot for your series of recent contribution, it will help improving guake for everyone to use!

Reply:
Just got a timeslot to contribute to the tool that I used daily :)